app/src/main/kotlin/com/flxrs/dankchat/preferences/notifications/highlights/HighlightsScreen.kt

```diff
@@ -410,6 +423,70 @@ private fun MessageHighlightItem(
                 )
             }
         }
+        val defaultColor = when(item.type) {
+            MessageHighlightItem.Type.Subscription, MessageHighlightItem.Type.Announcement -> ContextCompat.getColor(LocalContext.current, R.color.color_sub_highlight)
+            MessageHighlightItem.Type.ChannelPointRedemption                               -> ContextCompat.getColor(LocalContext.current, R.color.color_redemption_highlight)
+            MessageHighlightItem.Type.ElevatedMessage                                      -> ContextCompat.getColor(LocalContext.current, R.color.color_elevated_message_highlight)
+            MessageHighlightItem.Type.FirstMessage                                         -> ContextCompat.getColor(LocalContext.current, R.color.color_first_message_highlight)
+            MessageHighlightItem.Type.Username                                             -> ContextCompat.getColor(LocalContext.current, R.color.color_mention_highlight)
+            MessageHighlightItem.Type.Reply, MessageHighlightItem.Type.Custom              -> ContextCompat.getColor(LocalContext.current, R.color.color_mention_highlight)
+        }
+        val color = item.customColor ?: defaultColor
+        var showColorPicker by remember { mutableStateOf(false) }
+        var selectedColor by remember(color) { mutableIntStateOf(color) }
+        OutlinedButton(
```

Comment:
I wonder if it would make more sense to have a dedicated checkbox to enable "custom colors" and only showing the button then. I'm also not too much of a fan anymore to show the colored hex value of the color, maybe we can instead have a colored circle/shape and next to it a static button that just says "Change color" or something.
What do you think?

Reply:
Personally I don't like having the selector gated with a checkbox since using the default is equivalent to selecting the default color. I added a "default" button to the color picker to handle that case as well.

As for the shape and form of the button to open the picker, I have no strong opinion. I'm not particularly familiar with Android development so copying the existing code was the easiest way forward for me.